utils/feeLockHelper.ts

```diff
@@ -102,11 +102,47 @@ export async function calculateBuyPriceByMarket(
       JSON.stringify(
         await api.rpc.market.calculate_buy_price(
           poolId,
-          sellAssetId,
+          buyAssetId,
           sellAmount,
         ),
       ),
     ),
   );
   return value;
 }
+
+export async function calculateSellPriceWithImpact(
+  poolId: BN,
+  sellAssetId: BN,
+  sellAmount: BN,
+) {
+  const api = getApi();
+  const value = JSON.parse(
+    JSON.stringify(
+      await api.rpc.market.calculate_sell_price_with_impact(
+        poolId,
+        sellAssetId,
+        sellAmount,
+      ),
+    ),
+  );
+  return [stringToBN(value[0]), stringToBN(value[1])];
+}
+
+export async function calculateBuyPriceWithImpact(
```

Comment:
Add rpc to the  begining of the function name, so its easier to find it later, and move it to Market static class.

Reply:
Done